//指定proto3格式
syntax = "proto3";
package resourcecenter;
option java_multiple_files = true; //以外部类模式生成
option java_package = "com.szwdcloud.resource.edition"; //所在包名
option java_outer_classname = "RCEditionServiceProto"; //最外层类名称

import "RCCommon.proto";
import "google/protobuf/wrappers.proto";

//  教材和章节相关的
service RCEditionService {

    //  条件查询教材版本.沪教版,人教版...,
    rpc listEditionVersion (RequestEditionVersionList) returns (ResponseEditionVersionList) {}

    //  获取教材列表
    rpc listEdition (RequestEditionList) returns (ResponseEditionList) {}

    //  新增教材
    rpc insertEdition (RequestInsertEdition) returns (EmptyResp) {}

    //  修改教材
    rpc updateEdition (RequestUpdateEdition) returns (EmptyResp) {}

    //  删除教材
    rpc deleteEdition (RequestDeleteEdition) returns (BaseResponse) {}

    //  获取某个教材的章节列表
    rpc listChapter (RequestChapterList) returns (ResponseChapterList) {}

    //根据科目查教材版本(结果message EditionVersionVo)
    rpc listEditionBySubject (google.protobuf.Int32Value) returns (ListResp) {}
    //根据科目教材查年级(结果message EditionVo)
    rpc listGradeByEdition(SubjectEditionReq) returns(ListResp){}
    //根据科目教材年级查章节包含知识点(结果message StringValue)
    rpc listChapterByEdition(EditionGradeReq) returns(SingleResp){}
    //根据科目教材年级查章节不包含知识点(结果message StringValue)
    rpc listChapterNoPointByEdition(EditionGradeReq) returns(SingleResp){}
}

message SubjectEditionReq{
    //科目
    int32 subjectId = 1;
    //教材版本
    int32 editionVersionId = 2;
}

message EditionGradeReq{
    //科目
    int32 subjectId = 1;
    //教材版本
    int32 editionVersionId = 2;
    //年级
    int32 gradeId = 3;
    //上下学期
    int32 fascicleId = 4;
}

message RequestEditionVersionList {
    int32 id = 1;   //  主键id   非必填
    string name = 2;    //  名称模糊查询    非必填
}

message ResponseEditionVersionList {
    //操作标识（0成功 1失败）
    string code = 1;
    //操作结果
    string msg = 2;
    repeated EditionVersionVo datas = 3;
}

message EditionVersionVo{
    int32 id = 1;   //  主键id
    string name = 2;    //  名称
}

//  获取教材列表数据,所有字段都可不填
message RequestEditionList {
    int32 pageNo = 1;       //  页码   从1开始
    int32 pageSize = 2;     //  每页大小,默认10
    int32 gradeId = 3;      //  年级
    int32 editionVersionId = 4;        //  教材版本id,对应的是t_edition_version 表,教版,湘教版这样的
    int32 subjectId = 5;    //  科目
    int32 stageId = 6;      //  学段
}

//  删除教材
message RequestDeleteEdition{
    int32   id = 1;
}

//  修改教材
message RequestUpdateEdition {

    int32   id = 1;
    int32   stageId = 2;        //  学段id
    int32   gradeId = 3;        //  年级id
    int32   subjectId = 4;      //  科目id
    int32   editionVersionId = 5;        //  教材版本id,对应的是t_edition_version 表,教版,湘教版这样的
    int32   fascicleId = 6;     //  分册id
    string  coverUrl = 7;       //  封面地址
    string  gradeName = 8;      //  年级名
    string  editionVersionName = 9;   //  教材版本名,人教版,湘教版这样的
    string  chapterJson = 10;       //  章节列表json层级结构树
}

//  新增教材
message RequestInsertEdition {

    int32   stageId = 1;        //  学段id
    int32   gradeId = 2;        //  年级id
    int32   subjectId = 3;      //  科目id
    int32   editionVersionId = 4;        //  教材版本id,对应的是t_edition_version 表,教版,湘教版这样的
    int32   fascicleId = 5;     //  分册id
    string  coverUrl = 6;       //  封面地址
    string  gradeName = 7;      //  年级名
    string  editionVersionName = 8;   //  教材版本名,人教版,湘教版这样的
    string  chapterJson = 9;       //  章节列表json层级结构树
}

//  教材对象--给列表使用,不包含章节
message EditionVo {

    int32   id = 1;
    int32   stageId = 2;        //  学段id
    int32   gradeId = 3;        //  年级id
    int32   subjectId = 4;      //  科目id
    int32   editionVersionId = 5;        //  教材版本id,对应的是t_edition_version 表,教版,湘教版这样的
    int32   fascicleId = 6;     //  分册id
    string  coverUrl = 7;       //  封面地址
    string  gradeName = 8;      //  年级名
    string  editionVersionName = 9;   //  教材版本名,人教版,湘教版这样的
}

//  t_chapter   章节对象
message ChapterVo {
    int32    id = 1;   //   新增教材时不传这个
    string   chapterName = 2;       //  章节名称
    int32    chapterIndex = 3;      //  章节序号
    int32    editionId = 4;         //  对应的教材id  新增教材时不传
    int32   parentId = 5;           //  父章节id,一级目录推荐传0
}

// 获取某个教材的章节
message RequestChapterList {
    int32    editionId = 1;     //  教材id,必传的哈
    int32    parentId = 2;      //  非必填,传0或者不传则是获取一级节点,传-1则为获取所有
}

//  获取教材列表
message ResponseEditionList {
    //操作标识（0成功 1失败）
    string code = 1;
    //操作结果
    string msg = 2;

    int64 total = 3;    //  总数量
    int32 pages = 4;    //  总页数
    int32 pageNo = 5;   //  页码
    int32 pageSize = 6; //  每页大小
    repeated EditionVo datas = 7;
}

//  章节列表
message ResponseChapterList {
    string code = 1;    //  操作标识（0成功 1失败）
    string msg = 2;     //  操作结果
    repeated ChapterVo datas = 3;
}

message BaseResponse{
    string code = 1;    //  操作标识（0成功 1失败）
    string msg = 2;     //  操作结果
}